api/handlers.go

```diff
@@ -0,0 +1,84 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-recipe-api/recipe"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+//RecipeListHandler - get all recipes
+// USAGE: curl -X GET http://localhost:22300/recipes
+func (api *RecipeAPI) RecipeListHandler(w http.ResponseWriter, req *http.Request) {
+	var list recipe.List
+	if api.EnableMongoData {
+		var err error
+		list.Items, err = api.dataStore.Backend.GetRecipes()
+		if err != nil {
+			log.Event(req.Context(), "error getting recipes from mongo", log.ERROR, log.Error(err))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	} else {
+		list = recipe.FullList
+	}
+
+	c := len(list.Items)
+	list.Count = c
+	list.TotalCount = c
+	list.Limit = c
+
+	b, err := json.Marshal(list)
+	if err != nil {
+		log.Event(req.Context(), "error returned from json marshall", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(b)
+}
+
+//RecipeHandler - get recipe by ID
+// USAGE: curl -X GET http://localhost:22300/recipes/{id}
+func (api *RecipeAPI) RecipeHandler(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	recipeID := vars["id"]
+	logD := log.Data{"recipe_id": recipeID}
+
+	var r recipe.Response
+	found := false
+
+	for _, item := range recipe.FullList.Items {
```

Comment:
I think here we should check if we are using mongo or not, and then do the same that you did for the full list of recipes:
```
if api.EnableMongoData {
...
else {
    list = recipe.FullList 
}
```
and once you get the list, you can iterate items (regardless if they came from Mongo or the hardcoded list)

Reply:
This has been implemented in the PR below

Add “Get Specific Recipe from Mongo”
https://github.com/ONSdigital/dp-recipe-api/pull/114